packages/api/src/controllers/object-store.test.js

```diff
@@ -108,24 +108,23 @@ describe("controllers/object-stores", () => {
     });
 
     it("should get all object stores with admin authorization", async () => {
-      store.userId = adminUser.id;
       for (let i = 0; i < 4; i += 1) {
         const storeChangeId = JSON.parse(JSON.stringify(store));
+        storeChangeId.userId = uuid();
         storeChangeId.id = uuid();
         await server.store.create(storeChangeId);
         const res = await client.get(`/object-store/${storeChangeId.id}`);
         expect(res.status).toBe(200);
         const objStore = await res.json();
         expect(objStore.id).toEqual(storeChangeId.id);
+        // making sure url is coming back as null
+        expect(objStore.url).toEqual(null);
       }
 
-      const res = await client.get(`/object-store?userId=${adminUser.id}`);
+      const res = await client.get(`/object-store`);
       expect(res.status).toBe(200);
       const objStores = await res.json();
       expect(objStores.length).toEqual(4);
-
-      // making sure url is coming back as null
-      expect(objStores[0].url).toEqual(null);
```

Comment:
Note: I removed this check here because the write-only fields are actually _not_ being cleared out when it is an admin making the list query. Is this an intended behavior or is it a mistake to not do that in the list API? 

I ask especially because on this new push-target API I actually included an explicit check _not_ to filter the write-only fields when the caller was an admin ([here](https://github.com/livepeer/livepeer-com/pull/430/files#diff-a53538e800c8499cf95fcfdae3a64b30091960541997bd3db13f28149a68d520R69)).

So should I keep that, or should we always clear these fields? And if the latter, a follow-up: How does it work to actually get these fields in production? Like how does `go-livepeer` get the object-store URLs so it can write the video records there, since in the other object-store APIs this field is always cleared out? (it seems like the only one letting it go is this listing one).

Reply:
OK, for that last part I am understanding that `go-livepeer` calls [this `/stream/hook` API](https://github.com/livepeer/livepeer-com/blob/0bf3b3a12f34fe919d575d64507d7f76b6cc7983/packages/api/src/controllers/stream.js) from this [mediarserver.go code](https://github.com/livepeer/go-livepeer/blob/54ac51ba36d5dea7bcf0463823c0e90537b8880f/server/mediaserver.go) in which we return the object store URLs. I couldn't find any authorization in that `/hook` API, is that by design?

Would that be a place where we'd need to create some push-target logic to return the push-target URL as well? Or is there a separate hook that mist-server calls already?

Reply:
> Note: I removed this check here because the write-only fields are actually not being cleared out when it is an admin making the list query. Is this an intended behavior or is it a mistake to not do that in the list API?

Admins should be allowed to retrieve the actual fields in all cases, I think. So I guess it's a mistake not to do that in the get API?

> OK, for that last part I am understanding that go-livepeer calls this /stream/hook API from this mediarserver.go code in which we return the object store URLs. I couldn't find any authorization in that /hook API, is that by design?

Yeah, right now anyone with a stream ID (not a playbackID, importantly) could query that for the data. We discussed some potential ways to improve that workflow a long time ago, discussion here: https://github.com/livepeer/go-livepeer/issues/1278

> Would that be a place where we'd need to create some push-target logic to return the push-target URL as well? Or is there a separate hook that mist-server calls already?

It _would_ be, if go-livepeer was the thing doing the RTMP pushing. But we're going to have Mist take care of that via mist-api-connector, so no changes necessary there quite yet.

Reply:
Cool! Thanks for the discussion reference I'll take a look and will mark this as resolved.
FTR: I am leaving the code on the `push-target` controllers that hides the write-only fields _except_ for admin users.